### PR TITLE
Generated docs pipeline part two

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description

A follow on from #1310, this uses both a pinned docker image reference and triggers the platform pipeline to actually publish the documentation

#### Tests

Changed a source link in one ReadMe doc. Ran the pipeline & checked that it was reverted back. 

#### Documentation

Will update the Wiki for the release process.

### CODE REVIEW

File: ci/generate-docs.sh [hunk 1]
```diff
@@ -30,23 +31,30 @@ CLONE_URL="git@github.com:spatialos/gdk-for-unity.git"
...
         --namespace-filter=".*DeploymentLauncher" \
         --namespace-filter=".*PlaymodeTests" \
         --git-tag="${TAG}"
 
-# TODO: Trigger Filip's pipeline when that's live
+pushd "${OUTPUT_DIR}"
+    buildkite-agent artifact upload "**/*.md"
+popd
+
+function generate_step() {
```
Comment (jessicafalk) [inline ci/generate-docs.sh]: very small nit: I personally prefer to have the functions grouped together at the start

File: ci/generate-docs.sh [hunk 2]
```diff
@@ -30,23 +31,30 @@ CLONE_URL="git@github.com:spatialos/gdk-for-unity.git"
...
+    echo "steps:"
+    echo "  - trigger: platform-release-docs-readme"
+    echo "    label: Upload documentation"
+    echo "    build:"
+    echo "      env:"
+    echo "        bk_readme_project_name: gdk-for-unity"
```
Comment (jessicafalk) [inline ci/generate-docs.sh]: nit: think we generally use all uppercase for env variables

File: ci/generate-docs.sh [hunk 3]
```diff
@@ -4,8 +4,9 @@ set -e -u -x -o pipefail
 
 cd "$(dirname "$0")/../"
 
+DOCGEN_PIN="${DOCGEN_OVERRIDE:-v1.0}"
 CURRENT_DIR=$(pwd)
-TMP_DIR=$(mktemp -d)
+TMP_DIR=$(mktemp -d "${TMPDIR:-/tmp}/XXXXXXXXX")
```
Comment (jessicafalk) [inline ci/generate-docs.sh]: why so many X? what does that do?

Comment (zeroZshadow) [inline ci/generate-docs.sh]: xxxx?

File: ci/generate-docs.sh (hunk 2 above)
Comment (jamiebrynes7) [reply ci/generate-docs.sh]: I don't control this. Blame @filipVisko 😛

File: ci/generate-docs.sh (hunk 3 above)
Comment (jamiebrynes7) [reply ci/generate-docs.sh]: Controls how many random characters to generate in the path. 8 characters for sufficiently small chances of collision

Comment (jamiebrynes7) [reply ci/generate-docs.sh]: See https://github.com/spatialos/gdk-for-unity/pull/1328#discussion_r396390594

File: ci/generate-docs.sh (hunk 1 above)
Comment (jamiebrynes7) [reply ci/generate-docs.sh]: Sure, can move it

**Pull request merged**